eclipse/che
# Allow to replace backslash in usernames during first login

Signed-off-by: Sergii Kabashniuk <skabashniuk@redhat.com>



### What does this PR do?

Allow to replace backslash in usernames during first login
### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18404

### How to test this PR?
1. Deploy che without OpenShift OAuth integration `chectl server:start --installer=operator   --platform=openshift  -n che --che-operator-cr-patch-yaml cr.yaml` 
```yaml
spec:
  auth:
    openShiftoAuth: false
```
2. Enable user self-registration in Che Keycloak realm
![Знімок екрана 2020-11-23 о 16 23 26](https://user-images.githubusercontent.com/1614429/99973269-4167a800-2da8-11eb-8db1-f727d745e996.png)

3. Deploy che-server image `quay.io/skabashn/che-server:che18404`
4. Register a new user with a backslash.
![Знімок екрана 2020-11-23 о 16 18 00](https://user-images.githubusercontent.com/1614429/99973329-517f8780-2da8-11eb-8071-488a528b16d9.png)
4. Ensure that username in che doesn't contains backslash.
![Знімок екрана 2020-11-23 о 16 18 27](https://user-images.githubusercontent.com/1614429/99973377-5e9c7680-2da8-11eb-9394-d1c66cff2cc7.png)



### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
